webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteReducer.js

```diff
@@ -0,0 +1,58 @@
+import Immutable from 'seamless-immutable';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+} from './AutoCompleteConstants';
+
+const initialState = Immutable({
+  controller: null,
+  error: null,
+  results: [],
+  searchQuery: '',
+  status: null,
+  trigger: null,
+});
+
+export default (state = initialState, action) => {
+  const {
+    payload: {
+      controller,
+      error,
+      results,
+      searchQuery,
+      status,
+      trigger,
+    } = {}, type,
+  } = action;
```

Comment:
Just small cosmetic but I think it will look better as:
```js
  const {
     type,
     payload: {
       controller,
       error,
       results,
       searchQuery,
       status,
       trigger,
     } = {},
   } = action;
```